aspnetcore/signalr/dotnet-client/sample/SignalRChatClient/MainWindow.xaml.cs

```diff
@@ -14,7 +15,15 @@ public MainWindow()
 
             connection = new HubConnectionBuilder()
                 .WithUrl("http://localhost:53353/ChatHub")
-                .Build();        
+                .Build();
+
+            #region snippet_ClosedRestart
+            connection.Closed += async (error) =>
+            {
+                await Task.Delay(3000);
```

Comment:
To actually make this "work", we should consider having the same do a truely random delay (i.e. `random.Next(0, 5) * 1000` or something). The purpose of a random delay is to ensure that when the server returns, it doesn't get a barrage of clients reconnecting, so each client needs to pick a **different** delay.